src/Glpi/Controller/Rule/RuleListController.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2025 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Controller\Rule;
+
+use Glpi\Controller\AbstractController;
+use Glpi\Exception\Http\BadRequestHttpException;
+use Html;
+use Rule;
+use Session;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+use Symfony\Component\Routing\Attribute\Route;
+
+// @todo use final like in GenericListController ?
+final class RuleListController extends AbstractController
+{
+    private \RuleCollection $ruleCollection;
+
+    #[Route("/{class}/Search", name: "fixme", priority: -1)]
+    public function __invoke(Request $request): Response
+    {
+        $id =           (int) $request->get('id');
+        $reinit =       $request->get('reinit');
+        $replay_rule =  $request->get('replay_rule');
+        $reorder =      $request->request->get('action');
+        $reorder =      $request->get('action'); // pour debug, à supprimer
+        $item_class = $request->attributes->getString('class');
+
+        $this->ruleCollection = $this->getRuleCollectionInstanceFromRuleSubtype($item_class, (int) $_SESSION['glpiactive_entity']);
+
+        // dispatch
+        if (!is_null($reorder)) {
+            $this->moveRule($id, $reorder, (int) $request->request->get('condition'));
+
+            return new RedirectResponse($request->getPathInfo());
+        }
```

Comment:
IMHO, the `__invoke` method should have the less code as possible and be only a "proxy" method. It would permit, if we want to refactor the corresponding code, to only have to change the corresponding method.

```suggestion
        if ($request->request->has('action')) {
            return $this->handleAction($request);
        }
```